src/commands/mixu.ts

```diff
@@ -53,36 +77,66 @@ abstract class Mixu {
   }
 
   @SimpleCommand('mixu', { directMessage: false })
-  mixu(command: SimpleCommandMessage) {
-    if (!command.message.guild || !this.isMixuChannel(command.message.channel.id)) return
+  async mixuCommand(command: SimpleCommandMessage) {
+    if (!command.message.guild || !this.isMixuChannel(command.message.channel.id)) {
+      return
+    }
 
     const tiles = this.shuffle()
     const score = this.score(tiles)
 
-    if (score > this.bestScore) {
-      this.bestScore = score
-      this.bestMixuTiles = tiles
+    await this.findBestMixu()
+    if (score > this.bestMixu.score) {
+      this.bestMixu = {
+        id: '1',
+        owner: command.message.author.username,
+        tiles: tiles.join(','),
+        score,
+      }
+
+      await this.client.bestMixu
+        .update({
+          where: { id: '1' },
+          data: this.bestMixu,
+        })
+        .catch(console.error)
     }
 
     const text = this.stringify(tiles, command.message.guild)
-    command.message.channel.send(`:regional_indicator_m::regional_indicator_i::regional_indicator_x::regional_indicator_u:${text}`)
+    command.message.channel.send(
+      `:regional_indicator_m::regional_indicator_i::regional_indicator_x::regional_indicator_u:${text}`
+    )
   }
 
   @SimpleCommand('bestmixu', { directMessage: false })
-  bestMixu(command: SimpleCommandMessage) {
-    if (!command.message.guild || !this.isMixuChannel(command.message.channel.id) || this.bestMixuTiles.length === 0) {
+  async bestMixuCommand(command: SimpleCommandMessage) {
+    if (!command.message.guild || !this.isMixuChannel(command.message.channel.id)) {
       return
     }
 
-    const text = this.stringify(this.bestMixuTiles, command.message.guild)
+    await this.findBestMixu()
+    if (this.bestMixu.tiles.length !== 16) {
```

Comment:
This will always fail as tiles will be joined by commas so the length will instead be 37. So either store tiles as a number array or split on commas here

Reply:
looks like you can't store it as an array so just make sure to split it on this call. We are using the split array later on so maybe turn it into a const so we're not needlessly re-splitting

Reply:
Thank you for catching that! That was just silly of me, I tested the code and it all worked fine but then realized the return keyword was gone moments before committing and just added it back thinking nothing of it.